cl/sentinel/utils.go

```diff
@@ -114,3 +118,57 @@ func (s *Sentinel) oneEpochDuration() time.Duration {
 func (s *Sentinel) inMeshCap() float64 {
 	return float64((3600 * time.Second) / s.oneSlotDuration())
 }
+
+// updateENRAttSubnets calls the ENR to notify other peers their attnets preferences.
+func (s *Sentinel) updateENRAttSubnets(subnetIndex int, on bool) {
+	var subnetField []byte
+	if err := s.listener.LocalNode().Node().Load(enr.WithEntry(s.cfg.NetworkConfig.AttSubnetKey, subnetField)); err != nil {
+		return
+	}
+	subnetField = common.Copy(subnetField)
+	if len(subnetField) <= subnetIndex/8 {
+		return
+	}
+	if on {
+		subnetField[subnetIndex/8] |= 1 << (subnetIndex % 8)
+	} else {
+		subnetField[subnetIndex/8] &^= 1 << (subnetIndex % 8)
+	}
+	s.listener.LocalNode().Set(enr.WithEntry(s.cfg.NetworkConfig.AttSubnetKey, subnetField))
+}
+
+// updateENRSyncNets calls the ENR to notify other peers their attnets preferences.
+func (s *Sentinel) updateENRSyncNets(subnetIndex int, on bool) {
+	var subnetField []byte
+	if err := s.listener.LocalNode().Node().Load(enr.WithEntry(s.cfg.NetworkConfig.SyncCommsSubnetKey, subnetField)); err != nil {
+		return
+	}
+	subnetField = common.Copy(subnetField)
+	if len(subnetField) <= subnetIndex/8 {
+		return
+	}
+	if on {
+		subnetField[subnetIndex/8] |= 1 << (subnetIndex % 8)
+	} else {
+		subnetField[subnetIndex/8] &^= 1 << (subnetIndex % 8)
+	}
+	s.listener.LocalNode().Set(enr.WithEntry(s.cfg.NetworkConfig.SyncCommsSubnetKey, subnetField))
+}
+
+// updateENROnSubscription updates the ENR based on the subscription status to subnets/syncnets.
+func (s *Sentinel) updateENROnSubscription(topicName string, subscribe bool) {
+	s.metadataLock.Lock()
+	defer s.metadataLock.Unlock()
+	for i := 0; i < int(s.cfg.NetworkConfig.AttestationSubnetCount); i++ {
```

Comment:
A boring suggestion to remove this for loop. feel free to skip it if you think it's unnecessary or unworkable.
```golang
if strings.Contains(topicName, gossip.TopicNamePrefixBeaconAttestation) {
   subnet := getSubnet(topicName)
   s.updateENRAttSubnets(...)
}
```

Reply:
will look into thx

Reply:
oh this does not work unfortunately, because you also need to extract the index